src/gordon_gcp/plugins/event_consumer.py

```diff
@@ -30,15 +30,243 @@
+            'plugin': plugin,
+            'message': message
+        }
+        self.history_log.append(log_item)
+
+
+class GPSEventConsumerBuilder:
+    """Build and configure a :class:`GPSEventConsumer` object.
+
+    Args:
+        config (dict): Google Cloud Pub/Sub-related configuration.
+        success_channel (asyncio.Queue): queue to place a successfully
+            consumed message to be further handled by the ``gordon``
+            core system.
+        error_channel (asyncio.Queue): queue to place a message met
+            with errors to be further handled by the ``gordon`` core
+            system.
+        kwargs (dict): Additional keyword arguments to pass to the
+            event consumer.
+    """
+    def __init__(self, config, success_channel, error_channel, **kwargs):
+        self.config = config
+        self.success_channel = success_channel
+        self.error_channel = error_channel
+        self.kwargs = kwargs
+
+    def _validate_config(self):
+        errors = []
+        # req keys: keyfile, project, topic, subscription
+        # TODO (lynn): keyfile won't be required once we support other
+        #              auth methods
+        if not self.config.get('keyfile'):
+            msg = ('The path to a Service Account JSON keyfile is required to '
+                   'authenticate for Google Cloud Pub/Sub.')
+            errors.append(msg)
+
+        if not self.config.get('project'):
+            msg = ('The GCP project where Cloud Pub/Sub is located is '
+                   'required.')
+            errors.append(msg)
+
+        if not self.config.get('topic'):
+            msg = ('A topic for the client to subscribe to in Cloud Pub/Sub is '
+                   'required.')
+            errors.append(msg)
+
+        if not self.config.get('subscription'):
+            msg = ('A subscription for the client to pull messages from in '
+                   'Cloud Pub/Sub is required.')
+            errors.append(msg)
+
+        if errors:
+            exp_msg = 'Invalid configuration:\n'
+            for error in errors:
+                logging.error(error)
+                exp_msg += error + '\n'
+            raise exceptions.GCPConfigError(exp_msg)
+
+        topic_prefix = f'projects/{self.config["project"]}/topics/'
+        if not self.config.get('topic').startswith(topic_prefix):
+            self.config['topic'] = f'{topic_prefix}{self.config["topic"]}'
+
+        sub_prefix = f'projects/{self.config["project"]}/subscriptions/'
+        if not self.config.get('subscription').startswith(sub_prefix):
+            sub = f'{sub_prefix}{self.config["subscription"]}'
+            self.config['subscription'] = sub
+
+    def _init_auth(self):
+        # a publisher client can't be made with credentials if a channel is
+        # already made/provided, which is what happens when the emulator is
+        # running ಠ_ಠ
+        # See (https://github.com/GodsaogleCloudPlatform/
```

Comment:
I think you got some extra letters stuck in this URL

Reply:
looool